rfcs/theme-shape.md

```diff
@@ -0,0 +1,228 @@
+    background: theme.global.palette.grey[50], // background: var(--global-palette-grey-50)
+  }),
+});
+```
+
+```js
+/* After */
+import { makeStyles, globalColorTokens } from '@fluentui/react-components';
+
+const useStyles = makeStyles({
+  root: theme => ({
+    color: theme.alias.color.neutral.neutralStroke1, // color: var(--alias-color-neutral-neutralStroke1)
+    background: globalColorTokens.palette.grey[50], // background: '#ddd' 👈 value inlined/bound during babel-plugin/build
+  }),
+});
+```
+
+#### When the global token is bound to the styles?
+
+In the current implementation the global token is injected to the DOM as CSS variable by `FluentProvider` - the theme passed to the `FluentProvider` is the source of truth for all the tokens.
+
+In the proposed implementation, however, the global token value is inlined during the application bundling. That should not be a problem, we do not support changing global token values once the application is built and bundled.
+
+For components developed in isolation this will work as well as the tokens are bound only when the final application is bundled (not when the component is built).
+
+#### `babel-make-styles` consequences
+
+The global tokens are inlined when component styles are processed by `babel-make-styles` plugin.
+There are 3 different places, where the plugin can be run:
+
+1. FUI library build - this is not a problem as FUI library currently does not use any global color tokens in component styles (actually there are global color tokens used in `Badge` - this is a bug, to be reported and fixed)
+2. Build of a component developed in isolation - this would cause a problem but currently, the plugin is not used here
+3. Build of an application - this is the correct place where the global color tokens are supposed to be inlined.
+
+To be safe with the FUI library build (point 1 above) and fix the potential problem when running the babel plugin on a component (point 2), we will consider the following options (**all after beta**):
+
+- Add an option to the babel plugin to fail when a global color token is used (makes point 1 safe as it will protect us from accidentally using global color tokens).
+- Modify the babel plugin to keep global color tokens as imports instead of inlining them (fixes point 2 as it moves the binding from babel-plugin in component to bundler in the application).
+- Run the babel plugin only in the application build. Neither in the FUI library nor in the component build. Process also `node_modules` (to cover the FUI library and component) during the application build.
+- Remove the babel plugin. We can also decide that the babel plugin is not worth the complexity and remove it.
+
+### <a name="merge-global-alias"></a>Merge global and alias tokens to a single object
+
+Having the tokens split to global and alias tokens is definitely a valid concept for design. But does not mean anything to engineers.
```

Comment:
Except when debugging :)

Reply:
Could you please elaborate more on the kind of possible problems which would be easier to debug with the split.

Reply:
I mean debugging the inlining process might be difficult if there isn't traceability to the original value. Let's say something is red that shouldn't be.  I go to look at the CSS and I see var(--alias-error).  I follow to the variable and I see a value of #f00.  I don't know that the reason this is red is because it was inlined from --global-error.